src/app/resources/view-resources/resources-viewer.component.ts

```diff
@@ -40,7 +41,10 @@ export class ResourcesViewerComponent implements OnChanges, OnDestroy {
       .subscribe((resources) => {
         if (this.shownResourceId !== this.resourceId) {
           this.shownResourceId = this.resourceId;
-          this.setResource(resources.find((r: any) => r._id === this.resourceId));
+          this.resource = resources.find((r: any) => r._id === this.resourceId);
```

Comment:
Small difference, but rather than setting `resource` as a property on the component, you can remove those two extra lines and in the opening of `this.setResource` add:
```
if (!resource) {
  return;
}
```
This exits the function without running the code, so accomplishes the same goal with fewer lines.